### DISCUSSION TITLE
(Noob question) ValueError: No phonemizer found for language de.

### DISCUSSION THREAD

**Discussion by ninoppp** [coqui-ai/TTS#1490] [General]
Hey
I've just spun up TTS locally in a python 3.9.7 venv on an ubuntu vm.
However, when trying to install the [Thorsten model](https://github.com/thorstenMueller/deep-learning-german-tts), I get `ValueError: No phonemizer found for language de.`
More specifically:
`(ThorstenVoice) user@pc:~/ThorstenVoice$ tts-server --model_name tts_models/de/thorsten/tacotron2-DCA
 > tts_models/de/thorsten/tacotron2-DCA is already downloaded.
 > vocoder_models/de/thorsten/fullband-melgan is already downloaded.
 > Using model: Tacotron2
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log10
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:50.0
 | > mel_fmax:8000.0
 | > pitch_fmin:0.0
 | > pitch_fmax:640.0
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:50
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:/home/juan/.local/share/tts/tts_models--de--thorsten--tacotron2-DCA/scale_stats.npy
 | > base:10
 | > hop_length:256
 | > win_length:1024
Traceback (most recent call last):
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/tts/utils/text/tokenizer.py", line 192, in init_from_config
    DEF_LANG_TO_PHONEMIZER[config.phoneme_language], **phonemizer_kwargs
KeyError: 'de'

`The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/juan/ThorstenVoice/bin/tts-server", line 5, in <module>
    from TTS.server.server import main
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/server/server.py", line 102, in <module>
    synthesizer = Synthesizer(
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/utils/synthesizer.py", line 76, in __init__
    self._load_tts(tts_checkpoint, tts_config_path, use_cuda)
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/utils/synthesizer.py", line 113, in _load_tts
    self.tts_model = setup_tts_model(config=self.tts_config)
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/tts/models/__init__.py", line 13, in setup_model
    model = MyModel.init_from_config(config, samples)
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/tts/models/tacotron2.py", line 356, in init_from_config
    tokenizer, new_config = TTSTokenizer.init_from_config(config)
  File "/home/juan/ThorstenVoice/lib/python3.9/site-packages/TTS/tts/utils/text/tokenizer.py", line 195, in init_from_config
    raise ValueError(
ValueError: No phonemizer found for language de.
                        You may need to install a third party library for this language.
`
This is probably a really stupid question, but how can I install said third party library? Thanks for any help.

**Comment by p0p4k** [comment]
[gruut[cs,de,es,fr,it,nl,pt,ru,sv]~=2.0.0](https://github.com/coqui-ai/TTS/blob/0cf3265a4686d7e856bd472cdaf1572d61cab2b8/requirements.txt#L36)
 Did you install requirements.txt?

**Reply by ninoppp** [reply]
I followed [this](https://github.com/thorstenMueller/deep-learning-german-tts#quick-steps-for-synthesizing-voice) guide, so I did `pip install -U pip TTS`
When installing the requirements.txt, everything is already satisfied.

**Reply by thorstenMueller** [reply]
This model has been trained on `espeak` or `espeak-ng`. Is this installed and available on your system? Maybe you can try `pip install TTS==0.5.0` and give it another try.

**Reply by ninoppp** [reply]
@thorstenMueller installing espeak did the trick, thanks! Maybe it would be worth mentioning this requirement in the guide, for the ultra noobs like me :)

**Comment by erogol** [comment]
You need to install the backend you want to use. Espeak or Gruut. 

For Espeak run ```sudo apt install espeak``` on Ubuntu. We can't distribute espeak with TTS due to license conflicts.
